File: models/Collaboration.js
const mongoose = require('mongoose');

const deliverableSchema = new mongoose.Schema({
  type: { type: String, required: true }, 
  status: { type: String, enum: ['Pending', 'Submitted', 'Approved', 'Rejected'], default: 'Pending' },
  link: { type: String, default: '' }
});

const collaborationSchema = new mongoose.Schema({
  campaignId: { type: mongoose.Schema.Types.ObjectId, ref: 'Campaign', required: true },
  influencerId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  brandId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  deliverables: [deliverableSchema],
  paymentDetails: {
    amount: { type: Number, required: true },
    currency: { type: String, default: 'USD' },
    status: { type: String, enum: ['Pending', 'Paid'], default: 'Pending' }
  },
  status: { type: String, enum: ['Negotiating', 'Content Creation', 'Content Review', 'Completed'], default: 'Negotiating' },
  isNew: { type: Boolean, default: false }
}, { timestamps: true });

module.exports = mongoose.model('Collaboration', collaborationSchema);

File: backend-fastapi/app/models/collaboration.py
from pydantic import BaseModel
from typing import Optional, List


class DeliverableItem(BaseModel):
    type: str
    status: Optional[str] = "Pending"
    link: Optional[str] = ""


class CollaborationCreateRequest(BaseModel):
    campaignId: str
    influencerId: str
    deliverables: Optional[List[DeliverableItem]] = []
    paymentAmount: float
    currency: Optional[str] = "USD"


class CollaborationStatusRequest(BaseModel):
    status: str


class CollaborationDeliverablesRequest(BaseModel):
    deliverables: List[DeliverableItem]


class CollaborationPaymentRequest(BaseModel):
    status: str
    amount: Optional[float] = None
